classes/job/repositories/BaseRepository.php

```diff
@@ -19,23 +19,16 @@
 use Illuminate\Database\Eloquent\Builder;
 use Illuminate\Database\Eloquent\Collection;
 use Illuminate\Database\Eloquent\Model;
-use PKP\job\interfaces\Repository;
+use Illuminate\Pagination\LengthAwarePaginator;
 
-abstract class BaseRepository implements Repository
+abstract class BaseRepository
 {
-    /**
-     * @var Model
-     */
-    protected $model;
+    protected Model $model;
```

Comment:
Doable but do we need it ? also this will start pointing out error in the base repository class in the IDE if we moved ti `$model` prop out of it .

Reply:
Problem is redefining throws fatal error in `PHP8.1+` as such 
```
PHP Fatal error:  Type of PKP\job\repositories\FailedJob::$model must be Illuminate\Database\Eloquent\Model (as in class PKP\job\repositories\BaseRepository) in /Users/abir/Sites/code/omp-main/lib/pkp/classes/job/repositories/FailedJob.php on line 30
```

One way to handle it as remove the prop `$model` from `BaseRepository` and implement in both of the main repos . Then IDE will start pointing out error for all the methods in base repos as a lot of use such as `$this->model` . To overcome this , we can have a have an abstract method `getModel(): Model` but then we have to update all the use of `$this->model` to `$this->getModel()` in base repo .

Reply:
Hmm, I think a type hint where the type has the same parent class is supposed to be allowed. Isn't that what's happening with this? https://github.com/pkp/ojs/blob/main/classes/submission/Repository.php#L29

Reply:
For https://github.com/pkp/ojs/blob/main/classes/submission/Repository.php#L29 , there is not type defined but here in base repos , we are defining it as 
```php
protected Model $model;
```

and if try to define it in Job/FailedJob repos which extends base repo , we can not redefine it as 
```php
protected PKPJobModel $model;
// or
protected PKPFailedJobModel $model;
```
even though `PKPJobModel/PKPFailedJobModel` extends `Model`, it has to be exact same(can not be the extended or inherited one also) or `PHP8.1+` throws fatal error .  I don't remember the exact behaviour for `PHP8.0+`(probably a warning that in future it will throw fatal error) but this is the case also for no type or primitive type when extending in `PHP8.1+`

Reply:
Ah right, I didn't notice that we weren't defining a type for the map. That's probably why we didn't define a type for the map. Ok, leave it as-is for now.